sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseFacadeEjb.java

```diff
@@ -2415,20 +2415,32 @@ public void deleteCaseAsDuplicate(String caseUuid, String duplicateOfCaseUuid) t
 
 	@Override
 	@RolesAllowed(UserRight._CASE_ARCHIVE)
-	public void archive(String entityUuid, Date endOfProcessingDate) {
+	public void archive(String entityUuid, Date endOfProcessingDate, boolean includeContacts) {
 		super.archive(entityUuid, endOfProcessingDate);
+		if (includeContacts) {
+			List<String> caseContacts = contactService.getAllUuidsByCaseUuids(Collections.singletonList(entityUuid));
+			contactService.archive(caseContacts);
```

Comment:
Shouldn't the end of processing date be set here as well?

Reply:
no, for bulk archiving the endOfProcessingDate is calculated in the archive method from AbstractCoreAdoService